src/prefect/settings/models/client.py

```diff
@@ -87,6 +87,16 @@ class ClientSettings(PrefectBaseSettings):
         """,
     )
 
+    custom_headers: Dict[str, str] = Field(
```

Comment:
any reason not use the builtin?
```suggestion
    custom_headers: dict[str, str] = Field(
```

Reply:
There is a reason but I don't understand it! Loading settings fails with a pydantic error when I use the built-in.

Reply:
(discussing offline)